ts/src/pro/coinex.ts

```diff
@@ -240,70 +262,205 @@ export default class coinex extends coinexRest {
+        //         },
         //         "id": null
         //     }
         //
-        const params = this.safeValue (message, 'params', []);
-        const first = this.safeValue (params, 0, {});
-        this.balance['info'] = first;
-        const currencies = Object.keys (first);
-        for (let i = 0; i < currencies.length; i++) {
-            const currencyId = currencies[i];
+        const data = this.safeDict (message, 'data', {});
+        const balances = this.safeList (data, 'balance_list', []);
+        for (let i = 0; i < balances.length; i++) {
+            const entry = balances[i];
+            this.balance['info'] = entry;
+            const currencyId = this.safeString (entry, 'ccy');
             const code = this.safeCurrencyCode (currencyId);
-            const available = this.safeString (first[currencyId], 'available');
-            const frozen = this.safeString (first[currencyId], 'frozen');
             const account = this.account ();
-            account['free'] = available;
-            account['used'] = frozen;
+            account['free'] = this.safeString (entry, 'available');
+            account['used'] = this.safeString (entry, 'frozen');
             this.balance[code] = account;
             this.balance = this.safeBalance (this.balance);
         }
         const messageHash = 'balance';
         client.resolve (this.balance, messageHash);
     }
 
+    async watchMyTrades (symbol: Str = undefined, since: Int = undefined, limit: Int = undefined, params = {}): Promise<Trade[]> {
+        /**
+         * @method
+         * @name coinex#watchMyTrades
+         * @description watches information on multiple trades made by the user
+         * @see https://docs.coinex.com/api/v2/spot/deal/ws/user-deals
+         * @see https://docs.coinex.com/api/v2/futures/deal/ws/user-deals
+         * @param {string} [symbol] unified symbol of the market the trades were made in
+         * @param {int} [since] the earliest time in ms to watch trades
+         * @param {int} [limit] the maximum number of trade structures to retrieve
+         * @param {object} [params] extra parameters specific to the exchange API endpoint
+         * @returns {object[]} a list of [trade structures]{@link https://docs.ccxt.com/#/?id=trade-structure}
+         */
+        await this.loadMarkets ();
+        await this.authenticate ();
+        let market = undefined;
+        if (symbol !== undefined) {
+            market = this.market (symbol);
+            symbol = market['symbol'];
+        }
+        let type = undefined;
+        [ type, params ] = this.handleMarketTypeAndParams ('watchMyTrades', market, params);
+        const url = this.urls['api']['ws'][type];
+        const subscribedSymbols = [];
+        let messageHash = 'myTrades';
+        if (market !== undefined) {
+            messageHash += ':' + symbol;
+            subscribedSymbols.push (market['id']);
+        } else {
+            if (type === 'spot') {
```

Comment:
same thing here, I think we're not even resolving the any messageHash without symbol